tests/tekton-tasks_test.go

```diff
@@ -1,362 +1,113 @@
+			Entry("[test_id:TODO] role binding", &roleBindingResource),
+			Entry("[test_id:TODO] service account", &serviceAccountResource),
+		)
+
+		Context("resource deletion", func() {
+			DescribeTable("recreate after delete", expectRecreateAfterDelete,
+				Entry("[test_id:TODO] task", &taskResource),
+				Entry("[test_id:TODO] cluster role", &clusterRoleResource),
+				Entry("[test_id:TODO] role binding", &roleBindingResource),
+				Entry("[test_id:TODO] service account", &serviceAccountResource),
+			)
 		})
 	})
 
-	Context("resource deletion when CR is deleted", func() {
-		BeforeEach(func() {
-			strategy.CreateTTOIfNeeded()
-			waitUntilDeployed()
-		})
-
-		It("[test_id:TODO]operator should delete tekton-tasks", func() {
-			tto := getTekton()
-			deleteTekton(tto)
-			liveTasks := &pipeline.TaskList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveTasks,
-					client.MatchingLabels{
-						common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveTasks.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "there should be no tasks left")
-		})
-
-		It("[test_id:TODO]operator should delete service accounts", func() {
-			tto := getTekton()
-			deleteTekton(tto)
-			liveSA := &v1.ServiceAccountList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveSA,
-					client.MatchingLabels{
-						common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveSA.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "there should be no service accounts left")
-		})
-
-		It("[test_id:TODO]operator should delete cluster role", func() {
-			tto := getTekton()
-			deleteTekton(tto)
-			liveCR := &rbac.ClusterRoleList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveCR,
-					client.MatchingLabels{
-						common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveCR.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "there should be no cluster roles left")
-
-		})
-
-		It("[test_id:TODO]operator should delete role bindings", func() {
-			tto := getTekton()
-			deleteTekton(tto)
-			liveRB := &rbac.RoleBindingList{}
-			Eventually(func() bool {
-				err := apiClient.List(ctx, liveRB,
-					client.MatchingLabels{
-						common.AppKubernetesComponentLabel: string(common.AppComponentTektonTasks),
-						common.AppKubernetesManagedByLabel: common.AppKubernetesManagedByValue,
-					},
-				)
-				Expect(err).ToNot(HaveOccurred())
-				return len(liveRB.Items) == 0
-			}, tenSecondTimeout, time.Second).Should(BeTrue(), "there should be no role bindings left")
-		})
-	})
-	Context("multiple CRs deployed", func() {
+	Context("resource creation when DeployTektonTaskResources is set to false", func() {
```

Comment:
This context can be removed, because SSP operator does not touch tekton tasks if the feature gate is false.

Reply:
Done.